lib/rule.js

```diff
@@ -236,43 +236,84 @@ Rule.prototype = new (function () {
       , source
       , action
       , opts
-      , prereqs
       , parts
       , valid
       , src
       , tNs
       , createdTask
       , name = Task.getBaseTaskName(fullName)
       , nsPath = Task.getBaseNamespacePath(fullName)
-      , ns = this.ns.resolveNamespace(nsPath);
+      , ns = this.ns.resolveNamespace(nsPath)
+      , prereqs = this.prereqs.slice(); // Get a copy to work with
 
     pattern = this.pattern;
     source = this.source;
 
-    if (typeof source == 'string') {
-      src = Matcher.getSource(name, pattern, source);
+    // TODO: Write a utility function that appends a
+    // taskname to a namespace path
+    var completeNamespace = function(s){
+      return nsPath.split(':').filter(function (item) {
+        return !!item;
+      }).concat(s).join(':');
+    };
+
+    // Deal with the various things the source argument can be:
+    // 1: A function
+    if (typeof source == 'function') { 
```

Comment:
I'm not completely sold on the function idea, mostly because part of the idea here was to make Rules behave more like Tasks. Could you leave this change for a different PR, and maybe implement it for both Tasks and Rules?